Python/compile.c

```diff
@@ -2221,12 +2416,39 @@ compiler_class(struct compiler *c, stmt_ty s)
             compiler_exit_scope(c);
             return ERROR;
         }
+        if (asdl_seq_LEN(typeparams) > 0) {
+            if (!compiler_set_type_params_in_class(c, loc)) {
```

Comment:
Passing an argument to the class body would require modifying `__build_class__`, because currently it always calls the class body function with zero args. I suppose we could change that, but I feel it would end up more complicated than the current solution. It's not obvious how to make `__build_class__` take another optional argument, because all positional arguments (except the first two, which are the func and name) are bases, and all keyword arguments are passed on to the metaclass.

Reply:
Oh, yeah, I'd forgotten we'd have to work through `__build_class__`, that does sound too complicated to be worth it.